test/tests/parallel/set1/spam_virus/rspamd_full.bats

```diff
@@ -114,7 +121,7 @@ function teardown_file() { _default_teardown ; }
   _service_log_should_contain_string 'rspamd' 'S \(reject\)'
   _service_log_should_contain_string 'rspamd' 'reject "Gtube pattern"'
 
-  _print_mail_log_for_id "${MAIL_ID2}"
+  _print_mail_log_for_id "${MAIL_ID_SPAM}"
```

Comment:
## Context (_no action required_)

Curious I spun up DMS container and used swaks to compare output and logs:

### With Rspamd

```console
# Not a great rspamd setup but sufficient for reproduction:
$ docker run --rm -it \
  --hostname mail.example.test \
  --env PERMIT_DOCKER=container \
  --env ENABLE_RSPAMD=1 \
  --name dms \
  mailserver-testing:ci

# Separate terminal (_unless you run container above detached with `-d`):
$ swaks --server 0.0.0.0 \
  --to hello@example.test \
  --body 'XJS*C4JDBQADN1.NSBN3*2IDNEN*GTUBE-STANDARD-ANTI-UBE-TEST-EMAIL*C.34X' \
  --data 'Date: %DATE%\nTo: %TO_ADDRESS%\nFrom: %FROM_ADDRESS%\nSubject: test\n%NEW_HEADERS%\n%BODY%\n'

=== Trying 0.0.0.0:25...
=== Connected to 0.0.0.0.
<-  220 mail.example.test ESMTP
 -> EHLO mail.example.test
<-  250-mail.example.test
<-  250-PIPELINING
<-  250-SIZE 10240000
<-  250-ETRN
<-  250-ENHANCEDSTATUSCODES
<-  250-8BITMIME
<-  250 CHUNKING
 -> MAIL FROM:<root@mail.example.test>
<-  250 2.1.0 Ok
 -> RCPT TO:<hello@example.test>
<-  250 2.1.5 Ok
 -> DATA
<-  354 End data with <CR><LF>.<CR><LF>
 -> Date: Sun, 07 Jan 2024 01:21:08 +0000
 -> To: hello@example.test
 -> From: root@mail.example.test
 -> Subject: test
 ->
 -> XJS*C4JDBQADN1.NSBN3*2IDNEN*GTUBE-STANDARD-ANTI-UBE-TEST-EMAIL*C.34X
 ->
 ->
 -> .
<** 554 5.7.1 Gtube pattern
 -> QUIT
<-  221 2.0.0 Bye
=== Connection closed with remote host.
```

```
# Postfix logs:
Jan  7 01:21:08 mail postfix/smtpd[891]: CF26E13142: client=localhost[127.0.0.1]
Jan  7 01:21:08 mail postfix/cleanup[894]: CF26E13142: message-id=<20240107012108.CF26E13142@mail.example.test>
Jan  7 01:21:08 mail postfix/cleanup[894]: CF26E13142: milter-reject: END-OF-MESSAGE from localhost[127.0.0.1]: 5.7.1 Gtube pattern; from=<root@mail.example.test> to=<hello@example.test> proto=ESMTP helo=<mail.example.test>
```

The Postfix Queue ID remained constant throughout those logs, and you match the 2nd `postfix/cleanup` log for `milter-reject` and `5.7.1 Gtube pattern` (_which I guess is the source of "Gtube pattern" in earlier review feedback instead of GTUBE test string (wikipedia)_).

We can see that my earlier insights on the queue ID being returned to swaks output isn't entirely true, we lost that since the mail was rejected and got the rejection reason instead.

The initial part of that line with swaks also changed from `<-` to `<**`, which [swaks docs explain](https://github.com/jetmore/swaks/blob/073854d9ac87914487f1fd4ae6a87ce282375f73/doc/base.pod#output-options):

> **`<**`**
> Indicates an unexpected line sent by target server to swaks

vs

> **`<- `**
> Indicates an expected line sent by target server to swaks

---

Appending our own header with `--header 'Message-Id: <spam@example.test>'` doesn't help much:

```
Jan  7 01:36:21 mail postfix/smtpd[2387]: 85B99132DF: client=localhost[127.0.0.1]
Jan  7 01:36:21 mail postfix/cleanup[2390]: 85B99132DF: message-id=<spam@example.test>

Jan  7 01:36:21 mail postfix/cleanup[2390]: 85B99132DF: milter-reject: END-OF-MESSAGE from localhost[127.0.0.1]: 5.7.1 Gtube pattern; from=<root@mail.example.test> to=<hello@example.test> proto=ESMTP helo=<mail.example.test>
```

You'd need to get the match on that 2nd log line with `postfix/cleanup` that provides the queue ID and `Message-Id` header value via `message-id`.... Or be happy with that `swaks` output that responds with `5.7.1 Gtube pattern`?

---

### With SpamAssassin + Amavis

Now with `--env ENABLE_SPAMASSASSIN=1` instead of `--env ENABLE_RSPAMD=1` (_Amavis being enabled by default_), we get the following:

```console
$ swaks --server 0.0.0.0 \
  --to hello@example.test \
  --header 'Message-Id: <spam@example.test>'
  --body 'XJS*C4JDBQADN1.NSBN3*2IDNEN*GTUBE-STANDARD-ANTI-UBE-TEST-EMAIL*C.34X' \
  --data 'Date: %DATE%\nTo: %TO_ADDRESS%\nFrom: %FROM_ADDRESS%\nSubject: test\n%NEW_HEADERS%\n%BODY%\n'

=== Trying 0.0.0.0:25...
=== Connected to 0.0.0.0.
<-  220 mail.example.test ESMTP
 -> EHLO mail.example.test
<-  250-mail.example.test
<-  250-PIPELINING
<-  250-SIZE 10240000
<-  250-ETRN
<-  250-ENHANCEDSTATUSCODES
<-  250-8BITMIME
<-  250 CHUNKING
 -> MAIL FROM:<root@mail.example.test>
<-  250 2.1.0 Ok
 -> RCPT TO:<hello@example.test>
<-  250 2.1.5 Ok
 -> DATA
<-  354 End data with <CR><LF>.<CR><LF>
 -> Date: Sun, 07 Jan 2024 01:43:50 +0000
 -> To: hello@example.test
 -> From: root@mail.example.test
 -> Subject: test
 -> Message-Id: <spam@example.test>
 ->
 -> XJS*C4JDBQADN1.NSBN3*2IDNEN*GTUBE-STANDARD-ANTI-UBE-TEST-EMAIL*C.34X
 ->
 ->
 -> .
<-  250 2.0.0 Ok: queued as 792B6131D7
 -> QUIT
<-  221 2.0.0 Bye
=== Connection closed with remote host.
```

```
# Postfix / Amavis logs:
Jan  7 01:43:53 mail postfix/smtpd-amavis/smtpd[807]: E8339132AB: client=localhost[127.0.0.1]
Jan  7 01:43:53 mail postfix/cleanup[797]: E8339132AB: message-id=<spam@example.test>

Jan  7 01:43:53 mail amavis[803]: (00803-01) Passed SPAM {RelayedTaggedInbound,Quarantined}, [127.0.0.1]:57256 <root@mail.example.test> -> <hello@example.test>, quarantine: S/spam-SvjFaxs_mgzs.gz, Queue-ID: 792B6131D7, Message-ID: <spam@example.test>, mail_id: SvjFaxs_mgzs, Hits: 1003.5, size: 217, queued_as: E8339132AB, 823 ms
Jan  7 01:43:53 mail postfix/smtp-amavis/smtp[798]: 792B6131D7: to=<hello@example.test>, relay=127.0.0.1[127.0.0.1]:10024, delay=2.5, delays=0.06/0/1.6/0.82, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as E8339132AB)

Jan  7 01:43:53 mail dovecot: lmtp(hello@example.test)<809><Z8NYOdkBmmUpAwAAUi6ngw>: sieve: msgid=<spam@example.test>: fileinto action: stored mail into mailbox 'Junk'
```

This time the mail wasn't rejected, it got stored in Dovecot unlike with rspamd.
- As a result swaks has that expected `queued as $QUEUE_ID` line.
- Perhaps due to the internal sender with `swaks` instead of a separate container with an untrusted network/client, that'd behave differently, or Amavis / SA just isn't treating it rejection worthy spam? 🤷‍♂️  (_[`SPAMASSASSIN_SPAM_TO_INBOX=1`](https://docker-mailserver.github.io/docker-mailserver/v13.2/config/environment/#spamassassin_spam_to_inbox) default is why, [`SA_KILL` is ignored as a result](https://github.com/docker-mailserver/docker-mailserver/blob/0eb4ac771498e1eff91944e79c19c8605995bc07/target/scripts/startup/setup.d/security/misc.sh#L114)_)

Now getting rejection with ``SPAMASSASSIN_SPAM_TO_INBOX=0` avoids storing in Dovecot Junk, but didn't match the rejection behaviour of rspamd to `swaks` output:

```
<-  250 2.0.0 Ok: queued as F0A4A131DB
```

```
Jan  7 02:00:07 mail postfix/smtpd[806]: F0A4A131DB: client=localhost[127.0.0.1]
Jan  7 02:00:07 mail postfix/cleanup[809]: F0A4A131DB: message-id=<spam@example.test>

Jan  7 02:00:09 mail amavis[790]: (00790-01) Blocked SPAM {NoBounceInbound,Quarantined}, [127.0.0.1]:38908 <root@mail.example.test> -> <hello@example.test>, quarantine: x/spam-xdvY9cjuNelb.gz, Queue-ID: F0A4A131DB, Message-ID: <spam@example.test>, mail_id: xdvY9cjuNelb, Hits: 1003.5, size: 217, 1760 ms
Jan  7 02:00:09 mail postfix/smtp-amavis/smtp[810]: F0A4A131DB: to=<hello@example.test>, relay=127.0.0.1[127.0.0.1]:10024, delay=1.8, delays=0.06/0/0/1.8, dsn=2.5.0, status=sent (250 2.5.0 Ok, id=00790-01, DISCARD(bounce.suppressed))

Jan  7 02:00:09 mail postfix/qmgr[738]: F0A4A131DB: removed
```

Reply:
This is the reason the `--unchecked` flags is required when using mail that is rejected:

https://github.com/docker-mailserver/docker-mailserver/blob/f67362ed7479d7ca4d5a9e512af46cd7aed4972f/test/helper/sending.bash#L173-L178

You can see that I am querying the queue ID queue differently (see that I did there with the "q"s 😆).